### PR TITLE
Fix issue #142

### PR DESCRIPTION
**Pull request by llaville**
### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/composer-unused/symbol-parser/blob/main/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/composer-unused/symbol-parser/pulls) for the same update/change?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

See issue #142 for all details.

### CODE REVIEW

**Comment by llaville** [comment]
@icanhazstring Review requested

**Comment by icanhazstring** [comment]
@llaville will get to it asap.

File: src/Parser/PHP/ConsumedSymbolCollector.php [hunk 1]
```diff
@@ -56,6 +58,12 @@ public function enterNode(Node $node)
 
     public function getSymbolNames(): array
     {
+        foreach ($this->strategies as $strategy) {
```
Comment (icanhazstring) [inline src/Parser/PHP/ConsumedSymbolCollector.php:61]: Technically this implementation and fix works.

But logically composer-unused always uses all strategies, so this loop will add "some" performance impact as it iterates over all strategies.

See: https://github.com/composer-unused/composer-unused/blob/main/config/services.php#L68-L84

Comment (llaville) [reply src/Parser/PHP/ConsumedSymbolCollector.php:61]: I understand !

**Comment by llaville** [comment]
@ComiR gave the answer with is [context](https://github.com/composer-unused/symbol-parser/issues/142#issuecomment-1966048796). 
It's not related to `UseStrategy` as it point to us in is initial reporting #102 but on `ClassConstStrategy`. 

See https://github.com/composer-unused/symbol-parser/blob/0.2.2/src/Parser/PHP/Strategy/ClassConstStrategy.php#L22

The node `Expr_ClassConstFetch` is never proceed when magic keyword `self` is used !

So I'll close this PR that has non-sense now.

/cc @icanhazstring

**Pull request closed without merging**